xwiki-platform-core/xwiki-platform-resource/xwiki-platform-resource-temporary/src/main/java/org/xwiki/resource/temporary/internal/DefaultTemporaryResourceStore.java

```diff
@@ -75,31 +76,35 @@ public File createTemporaryFile(TemporaryResourceReference reference, InputStrea
     public File getTemporaryFile(TemporaryResourceReference reference) throws IOException
     {
         List<String> segments = new ArrayList<>();
-        segments.add("tmp");
-        segments.add(reference.getModuleId());
-        int safePathLength = 2;
+
         if (reference.getOwningEntityReference() != null) {
             for (EntityReference component : reference.getOwningEntityReference().getReversedReferenceChain()) {
                 segments.add(component.getName());
-                safePathLength++;
             }
         }
         if (!reference.getParameters().isEmpty()) {
             segments.add(String.valueOf(reference.getParameters().hashCode()));
-            safePathLength++;
         }
-        segments.addAll(reference.getResourcePath());
+
         String path = StringUtils.join(encode(segments), '/');
-        String safePath = StringUtils.join(encode(segments.subList(0, safePathLength)), '/');
+        String md5 = DigestUtils.sha1Hex(path);
```

Comment:
I don't mind the use of sha1, but then the variable name is not very accurate :)